### PR TITLE
Migrate style variation test to Playwright

### PR DESCRIPTION
**Pull request by JustinyAhin**
## What?
Migrate [`style-variation.test.js`](https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-tests/specs/editor/various/style-variation.test.js) to its Playwright version.

## Why?
See https://github.com/WordPress/gutenberg/pull/38570 for its background and rationale.

## How?
See [MIGRATION.md](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/MIGRATION.md) for migration steps.

## Testing Instructions
Run ` npm run test-e2e:playwright -- /test/e2e/specs/editor/various/style-variation.spec.js`.

### CODE REVIEW

File: test/e2e/specs/editor/various/style-variation.spec.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+test.describe( 'adding blocks', () => {
+	test( 'Should switch to the plain style of the quote block', async ( {
+		page,
+		pageUtils,
+	} ) => {
+		await pageUtils.createNewPost();
+
+		// Inserting a quote block
+		await pageUtils.insertBlock( { name: 'core/quote' } );
+		await page.keyboard.type( 'Quote content' );
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/style-variation.spec.js]: Nitpick: maybe we can inline the content directly inside the block? (Didn't test it)

```suggestion
		await pageUtils.insertBlock( { name: 'core/quote', attributes: { content: 'Quote content' } } );
```

File: test/e2e/specs/editor/various/style-variation.spec.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+		await pageUtils.clickBlockToolbarButton( 'Quote' );
+
+		const plainStyleButton = await page.locator(
+			'button[role="menuitem"]:has-text("Plain")'
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/style-variation.spec.js]: This should be using role selectors. (I know currently it lacks documentation on how to write role selectors. Hopefully, Playwright will soon publish their official doc.)

```suggestion
			'role=menuitem[name="Plain"i]'
```

File: test/e2e/specs/editor/various/style-variation.spec.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+		);
+		await plainStyleButton.click();
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/style-variation.spec.js]: We can just directly click on the element:

```suggestion
		await page.click( 'role=menuitem[name="Plain"i]' );
```

File: test/e2e/specs/editor/various/style-variation.spec.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+		// Check the content
+		const quoteContent = `<!-- wp:quote {"className":"is-style-plain"} -->\n<blockquote class="wp-block-quote is-style-plain"><p>Quote content</p></blockquote>\n<!-- /wp:quote -->`;
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/style-variation.spec.js]: Nitpick: let's inline this in `toBe()` and replace `\n` with actual line breaks for readability.

**Review by kevin940726 — COMMENTED** [review]
Nice work! Left some nitpicks, overall looks good.

We should delete the original test file though.

File: test/e2e/specs/editor/various/style-variation.spec.js (hunk 1 above)
Comment (JustinyAhin) [reply test/e2e/specs/editor/various/style-variation.spec.js]:
```
await pageUtils.insertBlock( {
  name: 'core/quote',
  attributes: { content: 'Quote content' },
} );
```
does not seem to work for the quote block.

File: test/e2e/specs/editor/various/style-variation.spec.js (hunk 2 above)
Comment (JustinyAhin) [reply test/e2e/specs/editor/various/style-variation.spec.js]: Thanks

File: test/e2e/specs/editor/various/style-variation.spec.js (hunk 4 above)
Comment (JustinyAhin) [reply test/e2e/specs/editor/various/style-variation.spec.js]: I get this error
CleanShot 2022-04-11 at 09 48 11@2x

when I replace it with line breaks.
CleanShot 2022-04-11 at 09 48 18@2x

File: test/e2e/specs/editor/various/style-variation.spec.js (hunk 1 above)
Comment (kevin940726) [reply test/e2e/specs/editor/various/style-variation.spec.js]: Looks like it should be:

```js
await pageUtils.insertBlock( { name: 'core/quote', attributes: { value: '<p>Quote content</p>' }, } );
```

File: test/e2e/specs/editor/various/style-variation.spec.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+		await page.keyboard.type( 'Quote content' );
+
+		await pageUtils.clickBlockToolbarButton( 'Quote' );
+
+		await page.click( 'role=menuitem[name="Plain"i]' );
+
+		// Check the content
+		const content = await pageUtils.getEditedPostContent();
+		expect( content ).toBe(
+			`<!-- wp:quote {"className":"is-style-plain"} -->\n<blockquote class="wp-block-quote is-style-plain"><p>Quote content</p></blockquote>\n<!-- /wp:quote -->`
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/style-variation.spec.js:24]: Nitpick:

```suggestion
			`<!-- wp:quote {"className":"is-style-plain"} -->
<blockquote class="wp-block-quote is-style-plain"><p>Quote content</p></blockquote>
<!-- /wp:quote -->`
```

File: test/e2e/specs/editor/various/style-variation.spec.js (hunk 4 above)
Comment (kevin940726) [reply test/e2e/specs/editor/various/style-variation.spec.js]: Yeah, we should remove the preceding spaces.

File: test/e2e/specs/editor/various/style-variation.spec.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+test.describe( 'adding blocks', () => {
+	test( 'Should switch to the plain style of the quote block', async ( {
+		page,
+		pageUtils,
+	} ) => {
+		await pageUtils.createNewPost();
+
+		// Inserting a quote block
+
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/style-variation.spec.js]: Let's remove this empty line.

```suggestion
```

**Review by kevin940726 — APPROVED** [review]
LGTM 👍

File: test/e2e/specs/editor/various/style-variation.spec.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+test.describe( 'adding blocks', () => {
+	test( 'Should switch to the plain style of the quote block', async ( {
+		page,
+		pageUtils,
+	} ) => {
+		await pageUtils.createNewPost();
+
+		// Inserting a quote block
+		await pageUtils.insertBlock( {
```
Comment (oandregal) [inline test/e2e/specs/editor/various/style-variation.spec.js:14]: Hey, as far as I understand, this PR changed the original test behaviour. Previously, we:

1. added a block
2. typed some content

These steps covered the case for which we wanted to make sure that users could type right after the block was inserted.

The playwright test does this instead:

1. adds a block with the contents we want

It doesn't look like we're testing for typing. Sorry, I'm not very familiar with playwright, but shouldn't we respect the existing behavior when migrating tests? @JustinyAhin @kevin940726

Comment (oandregal) [inline test/e2e/specs/editor/various/style-variation.spec.js:14]: Hey, as far as I understand, this PR changed the original test behaviour. Previously, we:

1. added a block
2. typed some content

These steps covered the case for which we wanted to make sure that users could type right after the block was inserted.

The playwright test does this instead:

1. adds a block with the contents we want

It doesn't look like we're testing for typing. Sorry, I'm not very familiar with playwright, but shouldn't we respect the existing steps when migrating tests? @JustinyAhin @kevin940726

Comment (oandregal) [reply test/e2e/specs/editor/various/style-variation.spec.js:14]: https://github.com/WordPress/gutenberg/pull/40466

**Pull request merged**